tests/integration/deploying/deploy_command_test.py

```diff
@@ -1,34 +1,32 @@
 from pathlib import Path
-from types import SimpleNamespace
 
 import pytest
-from pytest_mock import MockerFixture
-from starknet_py.net.models import StarknetChainId
 
-from protostar.commands.deploy.deploy_command import DeployCommand
+from tests.data.contracts import CONTRACT_WITH_CONSTRUCTOR
+from tests.integration.conftest import CreateProtostarProjectFixture
+from tests.integration.protostar_fixture import ProtostarFixture
+
+
+@pytest.fixture(name="protostar", scope="module")
+def protostar_fixture(create_protostar_project: CreateProtostarProjectFixture):
+    with create_protostar_project() as protostar:
+        protostar.create_files({"./src/main.cairo": CONTRACT_WITH_CONSTRUCTOR})
+        protostar.build_sync()
+        yield protostar
```

Comment:
this is duplicated code

Reply:
can we make a few of those, just with different scopes ?

Reply:
this would make things more clear and re-usable

Reply:
> this would make things more clear and re-usable

IMO the opposite would happen.
- This approach allows modifying project files before building the project, and such modification usually needs to be done per test suite / test case.
- The test suite would be harder to understand because it would operate on an environment defined in a different file or files.

Reply:
>  would operate on an environment defined in a different file or files.
that's usually how our fixtures work, and in terms of reusability it's totally fine. if anything, it makes code cleaner because we don't need to repeat ourselves. 
> such modification usually needs to be done per test suite / test case.
That's usually only adding some test files, or a tree structure right? We could parametrize the fixture with some predefined sources which would be copied to the tree.

Reply:
i can see this kind of code littering our tests all over - that's what i'm afraid of

Reply:
> that's usually how our fixtures work, and in terms of reusability it's totally fine. if anything, it makes code cleaner because we don't need to repeat ourselves.

It makes the code cleaner visually, but much more obscure logically and harder to maintain.

- colocation https://kentcdodds.com/blog/colocation#the-principle
- the principle of proximity https://kula.blog/posts/proximity_principle/

> That's usually only adding some test files, or a tree structure right? We could parametrize the fixture with some predefined sources which would be copied to the tree.

Not always. Sometimes there's a need to build the project after adding files (migration tests), sometimes there's no such need (`formatter_test.py`). You would have to parametrize the fixture per test case, which would cause duplication (consider `migrator_deploy_contract_cheatcode_test.py`).

Reply:
This approach won't scale well that's all im saying

Reply:
I'd say the opposite. It gives flexibility to adapt environment for test suite needs. Without that, test cases would have to adapt to the environment, which would result in lower quality tests. Fixture environments would become very rigid, because any change could break tests.